content/courses/advanced-ada/parts/data_types/shared_variable_control.rst

```diff
@@ -484,17 +484,641 @@ these array declarations are equivalent:
+
+             package Random_Sleep is new
+               Ada.Numerics.Discrete_Random
+                 (Sleep_Range);
+             use Random_Sleep;
+
+             G : Generator;
+          begin
+             Reset (G);
+             delay Duration (Random (G));
+          end Sleep_Some_Time;
+
+          Generate_Value : declare
+             use Lazy_Value_Random;
+
+             G             :         Generator;
+             Initial_Value :         Lazy_Value_Range;
+             Prior         : aliased Lazy_Value;
+          begin
+             Reset (G);
+             Initial_Value := Random (G);
+
+             if Atomic_Compare_And_Exchange
+               (Item    => Value,
+                Prior   => Prior,
+                Desired => Lazy_Value (Initial_Value))
+             then
+                Value_Modified_By := Task_ID;
+             end if;
+
+          end Generate_Value;
+
+          accept Stop do
+             Put_Line ("Current task ID:     "
+                       & Task_ID'Image);
+             Put_Line ("Value:               "
+                       & Value'Image);
+             Put_Line ("Modified by task ID: "
+                       & Value_Modified_By'Image);
+             Put_Line ("---------------------");
+          end Stop;
+       end A_Task;
+
+       Some_Tasks : array (1 .. 5) of A_Task;
+    begin
+       for I in Some_Tasks'Range loop
+          Some_Tasks (I).Start (I);
+       end loop;
+       for I in Some_Tasks'Range loop
+          Some_Tasks (I).Stop;
+       end loop;
+    end Show_Lazy_Initialization;
+
+In the :ada:`Show_Lazy_Initialization` procedure, the most important variable
+is :ada:`Value`, which is the variable we have to protect via a lock. In
+addition, we have the auxiliary :ada:`Value_Modified_By` variable, which
+indicates the ID of the task that initialized the :ada:`Value` variable.
+
+In this procedure, we also see two main
+:ref:`block statements <Adv_Ada_Block_Statements>`:
+
+- the block statement with the :ada:`Sleep_Some_Time` identifier, where we make
+  the task *sleep* for a random amount of time (in the :ada:`Sleep_Range`
+  range); and
+
+- the block statement with the :ada:`Generate_Value` identified, where we
+  generate a new value randomly and attempt to update the :ada:`Value` variable
+  (of :ada:`Lazy_Value` type).
+
+Let's discuss some details about the :ada:`Generate_Value` block statement. We
+start by declaring some variables. Here, it's important to highlight that the
+:ada:`Prior` variable is initialized with the default value
+(:ada:`Lazy_Value_Default_Value`). We then call the
+:ada:`Atomic_Compare_And_Exchange` function, and pass :ada:`Value` and
+:ada:`Prior` as actual parameters. We can have two possible outcomes:
+
+#. If :ada:`Value` hasn't been modified by a task yet, it will contain the
+   default value |mdash| which means that the value of the :ada:`Prior` and
+   :ada:`Value` variables matches. In this case, the call to
```

Comment:
A grammar nit. I think the above should be:
"which means that the value of the :ada:`Prior` and :ada:`Value` variables matches."
should be 
"which means that the values of the :ada:`Prior` and :ada:`Value` variables match."

Reply:
Yes, indeed, that sounds better. Thanks!